app/controllers/api/base_controller.rb

```diff
@@ -264,6 +244,68 @@ def action_permission
       end
     end
 
+    def parent_permission(child_permission)
+      case child_permission.to_s
+      when 'create', 'destroy'
+        'edit'
+      when 'edit', 'view'
+        'view'
+      else
+        raise ::Foreman::Exception.new(N_("unknown parent permission for %s"), "#{params[:controller]}##{child_permission}")
+      end
+    end
+
+    def parent_resource_details
+      classes_arr = params.map { |key, value| extract_resource_from_param(key, value) }
```

Comment:
The snippet was for inspiration I didn't test it :-) anyway here's fixed and tested version

```ruby
def parent_resource_details
  parent_name, parent_class, parent_id = nil
  params.find do |param, value|
    parent_id = value
    parent_name, parent_class = extract_resource_from_param(param)
  end

  return nil if parent_name.nil?
  parent_scope = scope_for(parent_class, :permission => "#{parent_permission(action_permission)}_#{parent_name.pluralize}")
  parent_scope = select_by_resource_id_scope(parent_scope, parent_class, parent_id)
  [parent_name, parent_scope]
end

def extract_resource_from_param(param)
  md = param.match(/(\w+)_id$/)
  md ? [md[1], resource_class_for(resource_name(md[1]))] : nil
end
```

with this it seems to work for me